### PR TITLE
[ci] chore: add some NPU's UT/ST

### PR DESCRIPTION
**Pull request by daikang6**
### What does this PR do?

> Use cases benchmarking GPUs，add some NPU's UT/ST
### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/verl-project/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

File: tests/models/test_engine.py [hunk 1]
```diff
@@ -273,7 +278,10 @@ def create_value_model(language_model_path, output_path):
 
 @pytest.mark.parametrize("strategy", ["fsdp", "fsdp2"])
 def test_critic_engine(strategy):
-    device_count = torch.cuda.device_count()
+    if device_name == "cuda":
```
Comment (wucong25) [inline tests/models/test_engine.py]: 可以调用verl/verl/utils/device.py里面的get_device_name，来避免if分支的判断

File: tests/models/test_engine.py [hunk 2]
```diff
@@ -336,12 +344,20 @@ def test_critic_engine(strategy):
...
-        hf_values = hf_output.logits[:, -response_length - 1 : -1, :].float().squeeze(-1).cpu()
+    if device_name == "cuda":
+        with torch.device("cuda"), torch.autocast(device_type="cuda", dtype=torch.bfloat16):
+            hf_model = AutoModelForTokenClassification.from_pretrained(
+                value_model_path, torch_dtype=torch.float32, attn_implementation="flash_attention_2"
+            )
+            hf_output = hf_model(input_ids.cuda(), attention_mask=attention_mask.cuda())
+            hf_values = hf_output.logits[:, -response_length - 1 : -1, :].float().squeeze(-1).cpu()
+    elif device_name == "npu":
+        with torch.device("npu"), torch.autocast(device_type="npu", dtype=torch.bfloat16):
```
Comment (wucong25) [inline tests/models/test_engine.py]: 这里也是，可以把device_type用变量来传入

File: tests/models/test_engine.py [hunk 3]
```diff
@@ -65,13 +65,15 @@
...
+from verl.utils.device import get_device_name
+device_name = get_device_name()
 
 
 def get_test_language_model(device_count):
     if device_count == 1:
         model = "~/models/HuggingFaceTB/SmolLM2-135M-Instruct"
     else:
-        model = "~/models/Qwen/Qwen2.5-0.5B"
+        model = "/mnt/weight/Qwen2.5-0.5B"
```
Comment (daikang6) [inline tests/models/test_engine.py]: 路径不对，不要用本地路径

File: tests/models/test_engine.py [hunk 4]
```diff
@@ -138,7 +140,10 @@ def create_training_config(model_type, strategy, device_count, model):
 @pytest.mark.parametrize("strategy", ["fsdp", "fsdp2", "megatron"])
 def test_actor_engine(strategy):
     ray.init()
-    device_count = torch.cuda.device_count()
+    if device_name == "cuda":
+        device_count = torch.cuda.device_count()
+    elif device_name == "npu":
```
Comment (daikang6) [inline tests/models/test_engine.py]: 可以调用verl/verl/utils/device.py里面的get_device_name，来避免if分支的判断

**Review by wucong25 — DISMISSED** [review]

File: tests/models/test_engine.py [hunk 5]
```diff
@@ -45,6 +45,7 @@
 from verl.single_controller.ray import RayClassWithInitArgs, RayResourcePool, RayWorkerGroup
 from verl.trainer.config import CheckpointConfig
 from verl.utils import tensordict_utils as tu
+from verl.utils.device import *
```
Comment (wucong25) [inline tests/models/test_engine.py]: 这里为啥需要import *， 最好用到了什么函数导入什么函数，按需导入

**Review by wucong25 — APPROVED** [review]
